external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchReadSupport.scala

```diff
@@ -138,6 +138,15 @@ private[kafka010] class KafkaMicroBatchReadSupport(
       fromOffsets = startPartitionOffsets ++ newPartitionInitialOffsets,
       untilOffsets = endPartitionOffsets,
       executorLocations = getSortedExecutorList())
+    offsetRanges.filter { range =>
```

Comment:
This was missing when we wrote Kafka source v2.

Reply:
@tdas Very good call. I found the fix here was not correct when writing a unit test.